File: client-unity/Assets/_Project/Scripts/util/InputUtils.cs
using UnityEngine;

public class InputUtils
{
	public static Vector3 ComputeMovementFromInput(bool upInput, bool leftInput, bool downInput, bool rightInput)
	{
		Vector3 answer = new Vector3();
		if (upInput)
		{
			answer += Vector3.forward;
		}
		if (leftInput)
		{
			answer += Vector3.left;
		}
		if (downInput)
		{
			answer += Vector3.back;
		}
		if (rightInput)
		{
			answer += Vector3.right;
		}
		return answer;
	}
}

File: server/EzySmashers-app-api/src/main/java/org/youngmonkeys/ezysmashers/app/utils/InputUtils.java
package org.youngmonkeys.ezysmashers.app.utils;

import com.tvd12.gamebox.math.Vec3;

public final class InputUtils {
    
    private InputUtils() {}

    public static Vec3 computeMovementFromInput(
        boolean upInput,
        boolean leftInput,
        boolean downInput,
        boolean rightInput
    ) {
        Vec3 answer = new Vec3();
        if (upInput) {
            answer.add(Vec3.FORWARD);
        }
        if (leftInput) {
            answer.add(Vec3.LEFT);
        }
        if (downInput) {
            answer.add(Vec3.BACKWARD);
        }
        if (rightInput) {
            answer.add(Vec3.RIGHT);
        }
        return answer;
    }
}
